packages/core/admin/admin/src/content-manager/hooks/useRelation/useRelation.js

```diff
@@ -41,16 +41,41 @@ export const useRelation = (cacheKey, { relation, search }) => {
     cacheTime: 0,
     enabled: relation.enabled,
     getNextPageParam(lastPage) {
-      // the API may send an empty 204 response
-      if (!lastPage || lastPage.pagination.page >= lastPage.pagination.pageCount) {
+      const isXToOneRelation = !lastPage?.pagination;
+
+      if (
+        !lastPage || // the API may send an empty 204 response
+        isXToOneRelation || // xToOne relations do not have a pagination
+        lastPage.pagination.page >= lastPage.pagination.pageCount
+      ) {
         return undefined;
       }
 
       // eslint-disable-next-line consistent-return
       return lastPage.pagination.page + 1;
     },
     select: (data) => ({
-      pages: data.pages.map((page) => ({ ...page, results: [...(page.results ?? [])].reverse() })),
+      ...data,
+      pages: data.pages.map((page) => {
+        let results = [];
+        const isXToOneRelation = !!page?.data;
+
+        // xToOne relations return an object, which we normalize so that relations
+        // always have the same shape
+        if (isXToOneRelation) {
+          results = [page.data];
+        } else if (page?.results) {
+          results = page.results.reverse();
+        }
+
+        const data = { ...page, results };
+
+        if (isXToOneRelation) {
+          delete data.data;
```

Comment:
AFAIK won't `page` only have `pagination` and `results`? therefore, we can just include `pagination` and avoid the direct mutation? 

*sponsored by the FP club*

Reply:
I see the boo-crew over there :p Changed it - I think you were right.